src/backend/fs.rs

```diff
@@ -22,6 +22,21 @@ pub async fn write_file(pk: Secp256k1PubKey, file_bytes: &[u8]) -> Result<Blake3
     let file_hash = Blake3Hash(blake3::keyed_hash(&x_only_pk.serialize(), file_bytes));
 
     trace!("TODO: Check if file catalog already exists");
+    let path = SYS_CFG
+        .volumes
+        .get(0)
+        .expect("First volume present")
+        .path
+        .join(CATALOG_DIR)
+        .join(file_hash.to_string());
+
+    trace!("Read catalog at {}", path.to_string_lossy());
+    let is_catalog = path.to_string_lossy().len();
+    //let is_catalog = std::path::Path::new(&path).exists();
```

Comment:
Can we do the exists check this way? The other way seems strange.

Reply:
Are you saying we should go with line (35) as the method of choice?